### ISSUE TITLE
GET /v3/versions mappings field doesn't match docs

### ISSUE THREAD

**Issue by anonymous123-code** [QuiltMC/update-quilt-meta#9]
The property `mappings` should contain a list of objects structured like this:
```json
{
  "gameVersion": "1.12.2",
  "separator": "+build.",
  "build": 1,
  "maven": "org.quiltmc:quilt-mappings:1.12.2+build.1",
  "version": "1.12.2+build.1",
  "hashed": "1.12.2+build.1"
}
```

But the response looks like this:
```json5
{
  //...
  "mappings": [
    "23w12a",
    "1.19.4",
    "1.19.2",
    "1.19.3",
    "1.19.4-rc3",
    "1.19.4-rc2",
    "1.19.4-rc1",
    "1.19.4-pre4",
    "1.19.4-pre3",
    "1.19.4-pre2",
    "1.19.4-pre1",
    "23w07a",
    "23w06a",
    "23w05a",
    "23w04a",
    "23w03a",
    "1.19.3-rc3",
    "1.19.3-rc1",
    "1.19.3-rc2",
    "1.19.3-pre3",
    "1.19.3-pre2",
    "1.19.3-pre1",
    "22w46a",
    "22w45a",
    "22w44a",
    "22w43a",
    "22w42a",
    "1.18.2",
    "1.19.2-rc2",
    "1.19.2-rc1",
    "1.19.1",
    "1.19.1-rc3",
    "1.19.1-rc2",
    "1.19.1-pre6",
    "1.19.1-pre5",
    "1.19.1-pre4",
    "1.19.1-pre3",
    "1.19.1-pre2",
    "1.19.1-rc1",
    "1.19.1-pre1",
    "22w24a",
    "1.19-rc2",
    "1.19",
    "1.19-rc1",
    "1.19-pre5",
    "1.19-pre4",
    "1.19-pre3",
    "1.19-pre2",
    "1.19-pre1",
    "22w19a",
    "22w18a",
    "22w16b",
    "22w17a",
    "22w16a",
    "22w15a",
    "22w14a"
  ]
  //...
}
```

Based on some skimming this might be a concurrency issue/race condition between `populateMappings()` and `populateQuiltMappings()`, both run `this.arrays.put("mappings", [...]);`.

**Comment by Akarys42** [comment]
I'm not exactly certain what the issue is. Looking at https://meta.quiltmc.org/v3/versions, everything seems to match?

**Comment by anonymous123-code** [comment]
Yes, you are right, but the rest of the response matched the docs, so I'm pretty sure I looked up the right route... And I mean if this is a race condition it kinda matches the expected behavior, right?

**Comment by Akarys42** [comment]
Here is the thing that I don't get, the `/v3/version` route in prod returns the correct objects, unlike what your issue says

**Comment by anonymous123-code** [comment]
It is right that current prod is correct, but I'm sure I saw a version of the response broken in the way described above. Given that it worked fine for you it seems likely that some caching was involved. I cant reproduce it anymore now.

However, both `populateMappings` and `populateQuiltMappings` in `Main` are executed asynchronously at 
https://github.com/QuiltMC/update-quilt-meta/blob/434e193fd2ba58074209bad2ef469ea842291835/src/main/java/org/quiltmc/Main.java#L89-L96

`populateMappings` reads the qm maven metadata, and generates a JsonArray with the beginning of the version name, which is the mc version, and puts that into `this.arrays` with the key "mappings"
https://github.com/QuiltMC/update-quilt-meta/blob/434e193fd2ba58074209bad2ef469ea842291835/src/main/java/org/quiltmc/Main.java#L148-L150

`populateQuiltMappings` builds a list of objects with the properties specified in the spec and puts that into `this.arrays` with the key "mappings"
https://github.com/QuiltMC/update-quilt-meta/blob/434e193fd2ba58074209bad2ef469ea842291835/src/main/java/org/quiltmc/Main.java#L160-L190

Then, after both are done, `this.arrays.get("mappings")` is put into the mappings property of the response.
So, from my understanding, there is a race condition, which either results in the correct format (`populateQuiltMappings` writes last), or the wrong format (`populateMappings` writes last)

**Comment by Akarys42** [comment]
That makes sense, thank you for clarifying

**Comment by anonymous123-code** [comment]
Im getting the wrong response again

**Comment by Akarys42** [comment]
Solved by 48d78a4
